assets/blocks/lesson-actions/lesson-actions-block/edit.js

```diff
@@ -1,24 +1,100 @@
+import { useState } from '@wordpress/element';
 import { InnerBlocks } from '@wordpress/block-editor';
+import { createBlock } from '@wordpress/blocks';
+import { useSelect, useDispatch } from '@wordpress/data';
 
-const innerBlocksTemplate = [
-	[ 'sensei-lms/button-complete-lesson', {} ],
-	[ 'sensei-lms/button-next-lesson', {} ],
-	[ 'sensei-lms/button-reset-lesson', {} ],
+import { LessonActionsBlockSettings } from './settings';
+
+const allowedBlocks = [
+	'sensei-lms/button-complete-lesson',
+	'sensei-lms/button-next-lesson',
+	'sensei-lms/button-reset-lesson',
 ];
+const innerBlocksTemplate = allowedBlocks.map( ( blockName ) => [
+	blockName,
+	{},
+] );
 
 /**
  * Edit lesson actions block component.
+ *
+ * @param {Object}   props
+ * @param {string}   props.clientId                Block ID.
+ * @param {Function} props.setAttributes           Block set attributes function.
+ * @param {Object}   props.attributes              Block attributes.
+ * @param {Object}   props.attributes.activeBlocks Active blocks, where the key is the block name.
  */
-const EditLessonActionsBlock = () => (
-	<InnerBlocks
-		allowedBlocks={ [
-			'sensei-lms/button-complete-lesson',
-			'sensei-lms/button-next-lesson',
-			'sensei-lms/button-reset-lesson',
-		] }
-		template={ innerBlocksTemplate }
-		templateLock="all"
-	/>
-);
+const EditLessonActionsBlock = ( {
+	clientId,
+	setAttributes,
+	attributes: { activeBlocks },
+} ) => {
+	const block = useSelect(
+		( select ) => select( 'core/block-editor' ).getBlock( clientId ),
+		[]
+	);
+	const { replaceInnerBlocks } = useDispatch( 'core/block-editor' );
+	const [ blocksAttributes, setBlocksAttributes ] = useState( {} );
+
+	/**
+	 * Toggle block.
+	 *
+	 * @param {string} blockName Block name.
+	 *
+	 * @return {Function} Function to toggle the block.
+	 */
+	const toggleBlock = ( blockName ) => ( on ) => {
```

Comment:
Do you think it would be better to put this on its own hook? 

We could also make the settings dumper if we created the `toggleBlock` callback here and updated them to have arguments `isResetActive` and `onResetChange` or similar.

Reply:
Good idea! It separated this concern, making the `EditLessonActionsBlock` cleaner, and making the setting dumber. =)
Thank you for the suggestion!